### PR TITLE
Allow "TwoFactor Nextcloud Notifications" to pull the state of the 2FA again

### LINKED ISSUE #551: Issue with NC v.22.2.0
Hi,
Today after update to the version 22.2.0, the 2FA APP Notification stop work.
The issue happened with different account's and different users. 
Some phones don't receive notification, and some receive, but after approval nothing happen. 
The same with approval by webApp notification (after approval nothing happen). 

The login keeps waiting permission, but nothing happen.

No logs were created. 

### Server configuration
**Operating system: Debian 10
**Web server: Apache
**Database: SQL
**PHP version: 8.0
**Nextcloud version: 22.2.0

Thank you for the help

### LINKED ISSUE #555: two-factor auth provider 'twofactor_webauthn' failed to load
I think this started after the update to 21.0.5.
The NC logs do not provide a useful message. The same is true for the browser's console.

### Steps to reproduce
1. With app twofactor_nextcloud_notification installed and option in the personal settings activated.
2. Log in.
3. Chose notofication as 2nd factor
4. Accept the notification on a second device

### Expected behaviour
The user should get logged in.

### Actual behaviour
Nothing happens.

### Server configuration
**Operating system**:
Ubuntu 

**Web server:**
nginx

**Database:**
PostgreSQL

**PHP version:**
7.4

**Nextcloud version:** (see Nextcloud admin page)
21.0.5 

### Client configuration
**Browser:**
Doesn't matter.

**Operating system:**
Doesn't matter

#### Nextcloud log (data/nextcloud.log)
```
{"reqId":"1Rb0UpCA7Bxutcjz4ID6","level":3,"time":"2021-10-02T14:55:15+02:00","remoteAddr":"192.168.xxx.xxx","user":"xxx","app":"core","method":"GET","url":"/login/selectchallenge?redirect_url=/ocs/v2.php/apps/twofactor_nextcloud_notification/api/v1/poll/J4IuhTP5ez4A88VRMdJRUSRe00rRwiHByx1qfzcA","message":"1 two-factor auth providers failed to load","userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:92.0) Gecko/20100101 Firefox/92.0","version":"21.0.5.1"}

```

#### Browser log
```
poll.js:26 promise rejected: TypeError: Cannot read properties of undefined (reading 'data')
    at challenge.js:37

```

### LINKED ISSUE #556: Sinsce Nextcloud Version 21.0.5 it does not authentificate correctly anymore
NC Server: 21.0.5
NC App Android: 3.17.0

Problem Description: If an admin is using the tfa with the app-login, it pops up correctly at the app. He improves, but nc does not redirect or authentificate, because if you reload you will be still stuck at the waiting for confirmation screen.

How to reproduce:
1.) Update to 21.0.5
2.) Authentificate as an admin
3.) Click approve at Android or Ios
4.) Hangs, no entry anymore

### PR DESCRIPTION
**Pull request by nickvergessen**
…A again

* Regression from #28725  which broken the https://github.com/nickv-nextcloud/twofactor_nextcloud_notification app
* Fixes https://github.com/nickv-nextcloud/twofactor_nextcloud_notification/issues/555
* Fixes https://github.com/nickv-nextcloud/twofactor_nextcloud_notification/issues/556
* Fixes https://github.com/nickv-nextcloud/twofactor_nextcloud_notification/issues/551

### CODE REVIEW

File: core/Middleware/TwoFactorMiddleware.php [hunk 1]
```diff
@@ -82,6 +83,11 @@ public function __construct(Manager $twoFactorManager, Session $userSession, ISe
 	 * @param string $methodName
 	 */
 	public function beforeController($controller, $methodName) {
+		if ($controller instanceof APIController && $methodName === 'poll') {
+			// Allow polling the twofactor nextcloud notifications state
+			return;
+		}
```
Comment (LukasReschke) [inline core/Middleware/TwoFactorMiddleware.php:89]: Does this work if the app is not locally installed? - If yes: LGTM.

**Review by icewind1991 — APPROVED** [review]

File: core/Middleware/TwoFactorMiddleware.php (hunk 1 above)
Comment (icewind1991) [reply core/Middleware/TwoFactorMiddleware.php:89]: afaik doing `instanceof` with a non existing class gracefully returns `false`

Comment (nickvergessen) [reply core/Middleware/TwoFactorMiddleware.php:89]: https://3v4l.org/cnR6V

Yes it works, same as ::class (which is what this uses internally I guess)

**Review by LukasReschke — APPROVED** [review]

**Pull request merged**